aten/src/ATen/cuda/detail/CUDAHooks.cpp

```diff
@@ -463,6 +466,131 @@ void CUDAHooks::deviceSynchronize(DeviceIndex device_index) const {
   c10::cuda::device_synchronize();
 }
 
+std::tuple<size_t, size_t, ptrdiff_t, uint64_t, bool, c10::IPCHandlePimpl>
```

Comment:
No, you've fundamentally misunderstood what exactly the pimpl is supposed to do here. You've still hard coded that this struct has to be three std::string. A true pimpl would have an abstract interface in c10 that has a virtual destructor and says nothing about what its internal state is, and then your CUDA implementation would subclass it and store precisely the data it had before.

But TBH, I preferred the old implementation (that was passing CI) because it is less abstract and there for easier for us to fix if we needed to in the future.